common/src/main/java/bio/terra/catalog/service/DatasetService.java

```diff
@@ -170,10 +175,13 @@ public void deleteMetadata(DatasetId datasetId) {
   public String getMetadata(DatasetId datasetId) {
     var dataset = datasetDao.retrieve(datasetId);
     ensureActionPermission(dataset, SamAction.READ_ANY_METADATA);
-    return new DatasetResponse(
-            dataset, getService(dataset.storageSystem()).getDataset(dataset.storageSourceId()))
-        .convertToObject()
-        .toString();
+    StorageSystemInformation information;
+    try {
+      information = getService(dataset.storageSystem()).getDataset(dataset.storageSourceId());
```

Comment:
This code looks kind of similar - What was happening before? Would this ForbiddenException bubble its way back up to the user?

Reply:
Yes, before this change the user would see the uncaught exception in the reply. It was a wrapped API Exception so it had the HTTP status that it got from TDR, which happens to be a 403.

The exception type here was wrong in a previous comment; all `DatarepoService` APIs throw `DatarepoException`.